### ISSUE TITLE
mel_fusion

### ISSUE THREAD

**Issue by amit-sofer** [LAION-AI/CLAP#62]
Hello
when creating the data and the mel fusion, you stacked the mels as:
mel_fusion = torch.stack([mel_chunk_front, mel_chunk_middle, mel_chunk_back, mel_shrink], dim=0)
meaning the global view is the last channel of the input.
but in the htsat patch embd you take the global view of the mels as the first channel:
global_x = x[:,0:1,:,:] 
did i miss any transitions you made? or did i miss understand? 
thank you for the grate work

**Comment by RetroCirce** [comment]
Thank you for posting this issue. It is definitely a very good question that we did not find before.
We are looking into this code and create a new branch for verifying it.   
If by confirmation, this is a problem, we will retrain the fusion model results and update in the arxiv (also in the camera ready version). But it should not affect much of our results, because the fusion is always done on 4 mel features.  
And technically the result can be better.

I will keep the update here. Please wait for us.

**Comment by RetroCirce** [comment]
Hi, after checking the code, we find that this is a problem that we wrongly put "shrink" in the "back" position.
This will affect our experiment results on fusion models but not cause any difference on w/o. fusion models.
We have updated our code, pretrained checkpoints, and results on the paper. The performance by using the correct fusion does not affect our performance of the fusion model (the results are different but still shows similar trends and outperform previous SoTAs).  We will release it in the camera ready version and the arxiv.

Now we have updated our repo, release our pip library, and a tutorial code to evaluate it in the ESC-50 dataset.  You can track our readme.md to use it.
